[Spring] resolves #16054, resolves #15088 add vendor extension x-spring-provide-args

dabdirb opened the pull request:
to fix #16054
to fix #15088 

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

dabdirb commented:
this PR is to fix issue 16054, by add vendor extenstion x-spring-provide-args

wing328 commented:
cc @cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02) @Zomzog (2022/09) @martin-mfg (2023/08)

martin-mfg commented:
Hi @dabdirb, thank you for the PR!

I have two comments:
- What's the purpose of this line in `petstore-with-spring-provide-args.yaml`?
   ```yaml
   #        - '@org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient("okta") org.springframework.security.oauth2.client.OAuth2AuthorizedClient authorizedClient'
   ```
   I tried uncommenting it, but then the generated code doesn't compile anymore.
- I think your new sample should be added to some pipeline, probably to `.github/workflows/samples-spring.yaml`.

dabdirb commented:
Hello, @martin-mfg 

* Whats the purpose, i want add use case of this vendor extension.
 ```
@org.springframework.security.oauth2.client.annotation.RegisteredOAuth2AuthorizedClient("okta") org.springframework.security.oauth2.client.OAuth2AuthorizedClient authorizedClient
```
This line is a use case when our service need access another server/service by webclient, with oauth token, we need a bean that autowired by spring framework, the oauth configuration, such as cliend-id and client-secret, is read from property file.
```
spring.security.oauth2.client.registration.okta.client-id=xxxxxxxxxx
spring.security.oauth2.client.registration.okta.client-secret=yyyyyyyyyyyyyyyyyyyy
```

please refer https://docs.spring.io/spring-security/reference/reactive/oauth2/client/authorized-clients.html

It can not be compile success because we should add spring security dependency in the POM.
I removed it from sample yaml to avoid confusing.

* Added  "samples/server/petstore/springboot-spring-provide-args"  into  .github/workflows/samples-spring.yaml

* "implicitHeadersRegex: ^Version.*" was removed

dabdirb commented on modules/openapi-generator/src/main/java/org/openapitools/codegen/VendorExtension.java:
```diff
@@ -7,6 +7,7 @@ public enum VendorExtension {
 
     X_IMPLEMENTS("x-implements", ExtensionLevel.MODEL, "Ability to specify interfaces that model must implements", "empty array"),
     X_SPRING_PAGINATED("x-spring-paginated", ExtensionLevel.OPERATION, "Add org.springframework.data.domain.Pageable to controller method. Can be used to handle page & size query parameters", "false"),
+    X_SPRING_PROVIDE_ARGS("x-spring-provide-args", ExtensionLevel.OPERATION, "Added spring provide arguments in controller", "empty array"),
```
@welshm , this description is better, i updated the VendorExtension.java
